src/AltinnCore/Runtime/RestControllers/ProcessController.cs

```diff
@@ -41,29 +66,445 @@ public class ProcessController : ControllerBase
+
+                if (startEventError != null)
+                {
+                    return startEventError;
+                }
+
+                instance = await StartProcessOfInstance(org, app, instanceOwnerId, instanceGuid, instance, startEvent);                
+            }
+
+            string currentTaskId = instance.Process.CurrentTask?.ElementId ?? instance.Process.StartEvent;
+
+            if (currentTaskId == null)
+            {
+                return Conflict($"Instance does not have valid currentTask");
+            }
+       
+            // do next until end event is reached or task cannot be completed.
+            int counter = 0;
+            do
+            {
+                List<string> nextElements = ProcessKeeper.NextElements(currentTaskId);
+
+                if (nextElements.Count > 1)
+                {
+                    return Conflict($"Cannot complete process. Multiple outgoing sequence flows detected from task {currentTaskId}. Please select manually among {nextElements}");
+                }
+
+                string nextElement = nextElements.First();
+
+                if (InstanceIsValid(instance))
+                {
+                    instance = await UpdateProcessStateToNextElement(org, app, instance, instanceOwnerId, instanceGuid, nextElement);                    
+                }
+                else
+                {
+                    return Conflict($"Instance is not valid in task {currentTaskId}. Automatic completeion of process is stopped");
+                }
+
+                currentTaskId = instance.Process.CurrentTask?.ElementId;
+            }
+            while (instance.Process.EndEvent == null || counter > MAX_ITERATIONS_ALLOWED);
+
+            if (counter > 1000)
+            {
+                return StatusCode(500, $"More than {counter} iterations detected in process. Possible loop. Fix app process definition!");
+            }
+
+            return Ok(instance.Process);
+        }
+
+        private async Task<Instance> StartProcessOfInstance(string org, string app, int instanceOwnerId, Guid instanceGuid, Instance instance, string validStartElement)
+        {
+            DateTime now = DateTime.UtcNow;
+
+            instance.Process = new ProcessState
+            {
+                Started = now,
+                StartEvent = validStartElement,
+            };
+            Instance updatedInstance = await instanceService.UpdateInstance(instance, app, org, instanceOwnerId, instanceGuid);
+            List<InstanceEvent> events = new List<InstanceEvent>
+            {
+                GenerateProcessChangeEvent("process:StartEvent", updatedInstance, validStartElement, now)
+            };
+
+            await DispatchEvents(org, app, events);
+
+            return updatedInstance;
+        }
+
+        private async Task<Instance> UpdateProcessStateToNextElement(string org, string app, Instance instance, int instanceOwnerId, Guid instanceGuid, string nextElementId)
```

Comment:
input parameters instanceOwnerId and instanceGuid are available in instance object

Reply:
Yes, but not in int and guid form. remember instance.Id = "{isntanceOwnerId}/{instanceGuid}